tests/test_generate.py

```diff
@@ -138,21 +144,44 @@ def verify_content(html_blob, tmpdir):
 def test_apidir(api_dir, tmpdir):
     test_bucket, credentials, storage_client = test_init()
 
-    bucket, yaml_blob, html_blob = setup_testdata(storage_client, test_bucket)
+    bucket, yaml_blob, html_blob = setup_testdata(
+        api_dir, storage_client, credentials, test_bucket
+    )
 
     # Test for api directory content
-    generate_and_upload(api_dir, storage_client, credentials, test_bucket)
+    run_generate(storage_client, credentials, test_bucket)
 
     verify_content(html_blob, tmpdir)
 
 
+def test_setup_docfx(yaml_dir):
+    test_bucket, credentials, storage_client = test_init()
+
+    bucket, yaml_blob, html_blob = setup_testdata(
+        yaml_dir, storage_client, credentials, test_bucket
+    )
+
+    tmp_path = pathlib.Path(tempfile.TemporaryDirectory(prefix="doc-pipeline.").name)
+    metadata_path = generate.setup_docfx(tmp_path, yaml_blob)
+
+    docfx_json_file = tmp_path.joinpath("docfx.json")
+    assert docfx_json_file.exists()
+    with open(docfx_json_file) as w:
```

Comment:
We could presumably have a golden docfx.json as well, instead of just checking one bit?

Reply:
I played around with this a bit, but didn't find a satisfactory way to test it that actually felt like it was testing more than the other `docfx.json` test we currently have. So, going to leave as-is unless you object.